Weighted best constant fit added to quantile distribution.

shandawg opened the pull request:
Fixed bug in quantile InitF and FitBestConstant to included weighted quantile.

Quantile loss function updated to use the a weighted quantile calculation. locationm.cpp & locationm.h files edited to include dAlpha as a input parameter, then tdist and laplace distributions updated accordingly to calculate weighted quantile at alpha = 0.5 (median).

shandawg commented:
Harry,
I didn't look at the open pull requests and just seen searchivarius' update on the same thing. But I feel like this bug fix should be updated.

harrysouthworth commented:
Hang on. The other pull request can't merge automatically, which is why it hasn't been.

Please can you take a look at the other pull request, make any edits (if any) to this one that you think are necessary, then let me know. At that point I'll merge it in.

Thanks,
Harry

shandawg commented:
Hi Harry,

Checked the other pull request and updated the gbm.fit.R to remove the warning.

Also renamed the mpLocM.Median function to mpLocM.weightedQuantile so as not to be confusing when alpha != 0.5.
